src/screentovnc.cpp

```diff
@@ -604,6 +613,181 @@ void ScreenToVnc::updateClientCursors(rfbScreenInfoPtr rfbScreen, bool emptyMous
     isEmptyMouse = emptyMouse;
 }
 
+void ScreenToVnc::uinputCreateKeyboardDevice()
+{
+  struct uinput_user_dev uud;
+
+  uinputKeyboardDeviceFD = open("/dev/uinput", O_WRONLY | O_NONBLOCK);
+
+  ioctl(uinputKeyboardDeviceFD, UI_SET_EVBIT, EV_KEY);
+  for (int i=0; i < 256; i++) {
```

Comment:
I wonder if there are some constants that could be used here or if we should define them explicitly based on what we actually use. At least [0 seems to be KEY_RESERVED](https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h#L75) and [codes after 248 are just undefined](https://github.com/torvalds/linux/blob/5833291ab6de9c3e2374336b51c814e515e8f3a5/include/uapi/linux/input-event-codes.h) plus [code 255 is said to be reserved](https://github.com/torvalds/linux/blob/5833291ab6de9c3e2374336b51c814e515e8f3a5/include/uapi/linux/input-event-codes.h#L340).

Reply:
yea, those 9 values could probably safely be removed, although there is no particular reason to do so. this appears to be the canonical way of saying "i am interested in all keyboard events".